pkg/application/inject/fuse/container_injector.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2021 The Fluid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fuse
+
+import (
+	"fmt"
+
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (s *Injector) mutateContainers(keyName types.NamespacedName, fuseContainerName string,
+	containers []corev1.Container, privileged bool,
+	datasetVolumeNames []string,
+	template *common.FuseInjectionTemplate,
+	volumeNamesConflict map[string]string) (result []corev1.Container,
+	injectFuseContainer bool) {
+
+	mountPropagationHostToContainer := corev1.MountPropagationHostToContainer
+	for _, container := range containers {
+		// Skip injection for injected container
+		if container.Name == fuseContainerName {
+			warningStr := fmt.Sprintf("===> Skipping injection because %v has injected %q sidecar already\n",
+				keyName, fuseContainerName)
+			log.Info(warningStr)
+			break
+		}
+
+		// Set mountPropagationHostToContainer to the dataset volume mount point, and set Injection true
+		for i, volumeMount := range container.VolumeMounts {
+			if utils.ContainsString(datasetVolumeNames, volumeMount.Name) {
+				if privileged {
+					container.VolumeMounts[i].MountPropagation = &mountPropagationHostToContainer
```

Comment:
`privileged` is not required for `HostToContainer` (https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation). I think maybe `privileged := true` is not necessary [here](https://github.com/fluid-cloudnative/fluid/pull/1849/files#diff-fb20bc8364b3bc378ae1ee0bd2eea7c4c24e8edf2bc83737cdd3567c01fb2711R144) for safety consideration.